phoenix-core/src/main/java/org/apache/phoenix/iterate/TableResultIterator.java

```diff
@@ -121,8 +170,21 @@ public synchronized void initScanner() throws SQLException {
                 this.scanIterator =
                         new ScanningResultIterator(htable.getScanner(scan), scanMetrics);
             } catch (IOException e) {
-                Closeables.closeQuietly(htable);
-                throw ServerUtil.parseServerException(e);
+                if(handleSplitRegionBoundaryFailureDuringInitialization) {
```

Comment:
Rather than have a handleSplitRegionBoundaryFailureDuringInitialization, would it work to always throw it and let the caller deal with it? When a variable name gets beyond 50 characters, it's a sign that it may be too special of a case. :-)